File: Syncfusion_Sample/Windows/Edit.Windows/Samples/2.0/Advanced Editor Functions/Find and Replace/CS/RemoveHistoryItem.cs
using System;
using System.Drawing;
using System.Collections;
using System.ComponentModel;
using System.Windows.Forms;

namespace FindReplaceDemo
{
	/// <summary>
	/// Summary description for RemoveHistoryItem.
	/// </summary>
	public class RemoveHistoryItem : System.Windows.Forms.Form
	{
		private System.Windows.Forms.ListBox HistoryList;
		private System.Windows.Forms.ListBox RemoveList;
		private System.Windows.Forms.Button btnOk;
		private System.Collections.ArrayList myArrayList;
		private System.Collections.ArrayList myRemoveList;
		private System.Windows.Forms.Button BtnCancel;
		private System.Windows.Forms.Button btnCancelRemoved;
		private System.Collections.ICollection collection;
		private System.Windows.Forms.Label label1;
		private System.Windows.Forms.Label label2;
		private System.Windows.Forms.Button BtnRemoveItem;
		/// <summary>
		/// Required designer variable.
		/// </summary>
		private System.ComponentModel.Container components = null;

		public RemoveHistoryItem(System.Collections.ArrayList ar)
		{
			//
			// Required for Windows Form Designer support
			//
			InitializeComponent();
			this.myArrayList=ar;
			//this.myRemoveList=new ArrayList();

			//
			// TODO: Add any constructor code after InitializeComponent call
			//
		}

		/// <summary>
		/// Clean up any resources being used.
		/// </summary>
		protected override void Dispose( bool disposing )
		{
			if( disposing )
			{
				if(components != null)
				{
					components.Dispose();
				}
			}
			base.Dispose( disposing );
		}

		#region Windows Form Designer generated code
		/// <summary>
		/// Required method for Designer support - do not modify
		/// the contents of this method with the code editor.
		/// </summary>
		private void InitializeComponent()
		{
            this.HistoryList = new System.Windows.Forms.ListBox();
            this.RemoveList = new System.Windows.Forms.ListBox();
            this.btnOk = new System.Windows.Forms.Button();
            this.BtnCancel = new System.Windows.Forms.Button();
            this.BtnRemoveItem = new System.Windows.Forms.Button();
            this.btnCancelRemoved = new System.Windows.Forms.Button();
            this.label1 = new System.Windows.Forms.Label();
            this.label2 = new System.Windows.Forms.Label();
            this.SuspendLayout();
            // 
            // HistoryList
            // 
            this.HistoryList.Location = new System.Drawing.Point(24, 32);
            this.HistoryList.Name = "HistoryList";
            this.HistoryList.Size = new System.Drawing.Size(136, 121);
            this.HistoryList.TabIndex = 0;
            // 
            // RemoveList
            // 
            this.RemoveList.Location = new System.Drawing.Point(208, 32);
            this.RemoveList.Name = "RemoveList";
            this.RemoveList.Size = new System.Drawing.Size(128, 121);
            this.RemoveList.TabIndex = 1;
            // 
            // btnOk
            // 
            this.btnOk.DialogResult = System.Windows.Forms.DialogResult.OK;
            this.btnOk.Location = new System.Drawing.Point(352, 48);
            this.btnOk.Name = "btnOk";
            this.btnOk.Size = new System.Drawing.Size(96, 28);
            this.btnOk.TabIndex = 2;
            this.btnOk.Text = "Remove";
            this.btnOk.Click += new System.EventHandler(this.btnOk_Click);
            // 
            // BtnCancel
            // 
            this.BtnCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel;
            this.BtnCancel.Location = new System.Drawing.Point(352, 96);
            this.BtnCancel.Name = "BtnCancel";
            this.BtnCancel.Size = new System.Drawing.Size(96, 28);
            this.BtnCancel.TabIndex = 3;
            this.BtnCancel.Text = "Cancel";
            // 
            // BtnRemoveItem
            // 
            this.BtnRemoveItem.Font = new System.Drawing.Font("Microsoft Sans Serif", 9.75F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.BtnRemoveItem.Location = new System.Drawing.Point(168, 48);
            this.BtnRemoveItem.Name = "BtnRemoveItem";
            this.BtnRemoveItem.Size = new System.Drawing.Size(32, 24);
            this.BtnRemoveItem.TabIndex = 4;
            this.BtnRemoveItem.Text = ">>";
            this.BtnRemoveItem.Click += new System.EventHandler(this.Remove_Click);
            // 
            // btnCancelRemoved
            // 
            this.btnCancelRemoved.Font = new System.Drawing.Font("Microsoft Sans Serif", 9.75F, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, ((byte)(0)));
            this.btnCancelRemoved.Location = new System.Drawing.Point(168, 88);
            this.btnCancelRemoved.Name = "btnCancelRemoved";
            this.btnCancelRemoved.Size = new System.Drawing.Size(32, 24);
            this.btnCancelRemoved.TabIndex = 5;
            this.btnCancelRemoved.Text = "<<";
            this.btnCancelRemoved.Click += new System.EventHandler(this.btnCancelRemoved_Click);
            // 
            // label1
            // 
            this.label1.Location = new System.Drawing.Point(24, 8);
            this.label1.Name = "label1";
            this.label1.Size = new System.Drawing.Size(100, 23);
            this.label1.TabIndex = 6;
            this.label1.Text = "Existing ";
            // 
            // label2
            // 
            this.label2.Location = new System.Drawing.Point(208, 8);
            this.label2.Name = "label2";
            this.label2.Size = new System.Drawing.Size(112, 24);
            this.label2.TabIndex = 7;
            this.label2.Text = "Remove List";
            // 
            // RemoveHistoryItem
            // 
            this.AutoScaleBaseSize = new System.Drawing.Size(5, 13);
            this.ClientSize = new System.Drawing.Size(464, 166);
            this.Controls.Add(this.label2);
            this.Controls.Add(this.label1);
            this.Controls.Add(this.btnCancelRemoved);
            this.Controls.Add(this.BtnRemoveItem);
            this.Controls.Add(this.BtnCancel);
            this.Controls.Add(this.btnOk);
            this.Controls.Add(this.RemoveList);
            this.Controls.Add(this.HistoryList);
            this.FormBorderStyle = System.Windows.Forms.FormBorderStyle.FixedDialog;
            this.MaximizeBox = false;
            this.Name = "RemoveHistoryItem";
            this.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent;
            this.Text = "RemoveHistoryItem";
            this.Load += new System.EventHandler(this.RemoveHistoryItem_Load);
            this.ResumeLayout(false);

		}
		#endregion

		private void RemoveHistoryItem_Load(object sender, System.EventArgs e)
		{			
			foreach(object item in this.myArrayList)
			{
				this.HistoryList.Items.Add(item);
			}			
		}

		private void Remove_Click(object sender, System.EventArgs e)
		{
            if (HistoryList.SelectedItem != null)
            {
                this.RemoveList.Items.Add(this.HistoryList.SelectedItem);
                this.HistoryList.Items.Remove(this.HistoryList.SelectedItem);
            }

		}
		
		private void btnOk_Click(object sender, System.EventArgs e)
		{			
			collection=this.RemoveList.Items;
			this.myRemoveList=new ArrayList(collection);
		}

		private void btnCancelRemoved_Click(object sender, System.EventArgs e)
		{
            if (RemoveList.SelectedItem != null)
            {
                this.HistoryList.Items.Add(this.RemoveList.SelectedItem);
                this.RemoveList.Items.Remove(this.RemoveList.SelectedItem);
            }
		}

		public System.Collections.ArrayList RemoveListItems
		{
			get
			{
				if(this.RemoveList.Items!=null)
					return this.myRemoveList;
				else
					return null;
			}
		}
	}
}

File: Syncfusion_Sample/Windows/Edit.Windows/Samples/2.0/Advanced Editor Functions/Find and Replace/VB/RemoveHistoryItem.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Drawing
Imports System.Collections
Imports System.ComponentModel
Imports System.Windows.Forms

Namespace FindReplaceDemo
	''' <summary>
	''' Summary description for RemoveHistoryItem.
	''' </summary>
	Public Class RemoveHistoryItem : Inherits System.Windows.Forms.Form
		Private HistoryList As System.Windows.Forms.ListBox
		Private RemoveList As System.Windows.Forms.ListBox
		Private WithEvents btnOk As System.Windows.Forms.Button
		Private myArrayList As System.Collections.ArrayList
		Private myRemoveList As System.Collections.ArrayList
		Private BtnCancel As System.Windows.Forms.Button
		Private WithEvents btnCancelRemoved As System.Windows.Forms.Button
        Private collection As System.Collections.ICollection
		Private label1 As System.Windows.Forms.Label
		Private label2 As System.Windows.Forms.Label
		Private WithEvents BtnRemoveItem As System.Windows.Forms.Button
		''' <summary>
		''' Required designer variable.
		''' </summary>
		Private components As System.ComponentModel.Container = Nothing

		Public Sub New(ByVal ar As System.Collections.ArrayList)
			'
			' Required for Windows Form Designer support
			'
			InitializeComponent()
			Me.myArrayList=ar
			'this.myRemoveList=new ArrayList();

			'
			' TODO: Add any constructor code after InitializeComponent call
			'
		End Sub

		''' <summary>
		''' Clean up any resources being used.
		''' </summary>
		Protected Overrides Overloads Sub Dispose(ByVal disposing As Boolean)
			If disposing Then
				If Not components Is Nothing Then
					components.Dispose()
				End If
			End If
			MyBase.Dispose(disposing)
		End Sub

		#Region "Windows Form Designer generated code"
		''' <summary>
		''' Required method for Designer support - do not modify
		''' the contents of this method with the code editor.
		''' </summary>
		Private Sub InitializeComponent()
            Me.HistoryList = New System.Windows.Forms.ListBox
            Me.RemoveList = New System.Windows.Forms.ListBox
            Me.btnOk = New System.Windows.Forms.Button
            Me.BtnCancel = New System.Windows.Forms.Button
            Me.BtnRemoveItem = New System.Windows.Forms.Button
            Me.btnCancelRemoved = New System.Windows.Forms.Button
            Me.label1 = New System.Windows.Forms.Label
            Me.label2 = New System.Windows.Forms.Label
            Me.SuspendLayout()
            '
            'HistoryList
            '
            Me.HistoryList.Location = New System.Drawing.Point(24, 32)
            Me.HistoryList.Name = "HistoryList"
            Me.HistoryList.Size = New System.Drawing.Size(136, 121)
            Me.HistoryList.TabIndex = 0
            '
            'RemoveList
            '
            Me.RemoveList.Location = New System.Drawing.Point(208, 32)
            Me.RemoveList.Name = "RemoveList"
            Me.RemoveList.Size = New System.Drawing.Size(128, 121)
            Me.RemoveList.TabIndex = 1
            '
            'btnOk
            '
            Me.btnOk.DialogResult = System.Windows.Forms.DialogResult.OK
            Me.btnOk.Location = New System.Drawing.Point(352, 48)
            Me.btnOk.Name = "btnOk"
            Me.btnOk.Size = New System.Drawing.Size(96, 28)
            Me.btnOk.TabIndex = 2
            Me.btnOk.Text = "Remove"
            '
            'BtnCancel
            '
            Me.BtnCancel.DialogResult = System.Windows.Forms.DialogResult.Cancel
            Me.BtnCancel.Location = New System.Drawing.Point(352, 96)
            Me.BtnCancel.Name = "BtnCancel"
            Me.BtnCancel.Size = New System.Drawing.Size(96, 28)
            Me.BtnCancel.TabIndex = 3
            Me.BtnCancel.Text = "Cancel"
            '
            'BtnRemoveItem
            '
            Me.BtnRemoveItem.Font = New System.Drawing.Font("Microsoft Sans Serif", 9.75!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
            Me.BtnRemoveItem.Location = New System.Drawing.Point(168, 48)
            Me.BtnRemoveItem.Name = "BtnRemoveItem"
            Me.BtnRemoveItem.Size = New System.Drawing.Size(32, 24)
            Me.BtnRemoveItem.TabIndex = 4
            Me.BtnRemoveItem.Text = ">>"
            '
            'btnCancelRemoved
            '
            Me.btnCancelRemoved.Font = New System.Drawing.Font("Microsoft Sans Serif", 9.75!, System.Drawing.FontStyle.Bold, System.Drawing.GraphicsUnit.Point, CType(0, Byte))
            Me.btnCancelRemoved.Location = New System.Drawing.Point(168, 88)
            Me.btnCancelRemoved.Name = "btnCancelRemoved"
            Me.btnCancelRemoved.Size = New System.Drawing.Size(32, 24)
            Me.btnCancelRemoved.TabIndex = 5
            Me.btnCancelRemoved.Text = "<<"
            '
            'label1
            '
            Me.label1.Location = New System.Drawing.Point(24, 8)
            Me.label1.Name = "label1"
            Me.label1.Size = New System.Drawing.Size(100, 23)
            Me.label1.TabIndex = 6
            Me.label1.Text = "Existing "
            '
            'label2
            '
            Me.label2.Location = New System.Drawing.Point(208, 8)
            Me.label2.Name = "label2"
            Me.label2.Size = New System.Drawing.Size(112, 24)
            Me.label2.TabIndex = 7
            Me.label2.Text = "Remove List"
            '
            'RemoveHistoryItem
            '
            Me.AutoScaleBaseSize = New System.Drawing.Size(5, 13)
            Me.ClientSize = New System.Drawing.Size(464, 166)
            Me.Controls.Add(Me.label2)
            Me.Controls.Add(Me.label1)
            Me.Controls.Add(Me.btnCancelRemoved)
            Me.Controls.Add(Me.BtnRemoveItem)
            Me.Controls.Add(Me.BtnCancel)
            Me.Controls.Add(Me.btnOk)
            Me.Controls.Add(Me.RemoveList)
            Me.Controls.Add(Me.HistoryList)
            Me.MaximizeBox = False
            Me.Name = "RemoveHistoryItem"
            Me.ShowIcon = False
            Me.StartPosition = System.Windows.Forms.FormStartPosition.CenterParent
            Me.Text = "RemoveHistoryItem"
            Me.ResumeLayout(False)

        End Sub
		#End Region

        Private Sub RemoveHistoryItem_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
            Dim item As Object
            For Each item In Me.myArrayList
                Me.HistoryList.Items.Add(item)
            Next item

        End Sub

        Private Sub Remove_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles BtnRemoveItem.Click
            If Not HistoryList.SelectedItem Is Nothing Then
                Me.RemoveList.Items.Add(Me.HistoryList.SelectedItem)
                Me.HistoryList.Items.Remove(Me.HistoryList.SelectedItem)
            End If
        End Sub

        Private Sub btnOk_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnOk.Click
            collection = Me.RemoveList.Items
            Me.myRemoveList = New ArrayList(collection)
        End Sub

        Private Sub btnCancelRemoved_Click(ByVal sender As Object, ByVal e As System.EventArgs) Handles btnCancelRemoved.Click
            If Not RemoveList.SelectedItem Is Nothing Then
                Me.HistoryList.Items.Add(Me.RemoveList.SelectedItem)
                Me.RemoveList.Items.Remove(Me.RemoveList.SelectedItem)
            End If
        End Sub

        Public ReadOnly Property RemoveListItems() As System.Collections.ArrayList
            Get
                If Not Me.RemoveList.Items Is Nothing Then
                    Return Me.myRemoveList
                Else
                    Return Nothing
                End If
            End Get
        End Property
    End Class
End Namespace
